### ISSUE TITLE
No way to provide a custom schema with `__dask_capable__=True`

### ISSUE THREAD

**Issue by veprbl** [scikit-hep/coffea#867]
**Describe the bug**
This code implements external traits for a limited set of Schema's:
https://github.com/CoffeaTeam/coffea/blob/03ea8a52d76467bac321681c034c642fd03bd7ca/src/coffea/nanoevents/factory.py#L284-L310
providing anything unexpected results in attempting to evaluate `dict(None)` in
https://github.com/CoffeaTeam/coffea/blob/03ea8a52d76467bac321681c034c642fd03bd7ca/src/coffea/nanoevents/factory.py#L314

**To Reproduce**
```python
class MySchema(BaseSchema):
    pass

iterative_run = processor.Runner(
    executor=processor.IterativeExecutor(compression=None),
    schema=MySchema,
    maxchunks=10,
)

out = iterative_run(
    {"bla": files},
    treename="events",
    processor_instance=MyProcessor(),
)
```

**Expected behavior**

**Output**
```
  File "prefix/lib/python3.10/site-packages/coffea/nanoevents/factory.py", line 314, in from_root
    behavior=dict(behavior),
TypeError: 'NoneType' object is not iterable
```

**Comment by lgray** [comment]
Hey - this one is known and is going to be fixed! 
There are a few larger issues to take care of first, before getting to this one. 
Coffea 2023 is currently a release candidate and there can be problems!

A work around you can do now is to stick your schema class into `coffea/nanoevents/schemas` wherever you have coffea installed at.

I'd also suggest *not* using the runner/executor interface and use `dask.compute` along side native dask executors (use `sync` (same as iterative) or `processes` (same as futures), uproot does not play nicely with `threads`.
Nominally, `dask.compute` with the `distributed` package is the same as the dask executor.

WorkQueue will migrate to TaskVine, and you can use parsl via TaskVine. Eventually parsl will have its own dask taskgraph frontend as well!

If you're feeling adventurous you can try using ray.

**Comment by veprbl** [comment]
Thank you for suggestions. Sure, I could use dask, but what would be the "right" way to merge results from different processors then?

**Comment by lgray** [comment]
It does it automatically for you, it's part of dask-histogram and dask itself! (processor, as a concept, is sort of ill-defined when you are using dask, we need to come up with better concepts)

**Comment by lgray** [comment]
@veprbl This should be fixed now in #882 - could you give it a try?
